### PR TITLE
feat(curriculum): add functions calculator workshop

### PR DESCRIPTION
**Pull request by jdwilkin4**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/workshop-calculator-app/66cc0a9e06e00b75d6782be9.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+```js
+function myFunction(){
+  return 'Hello World';
+}
+
+// function call
+myFunction();
+```
+
+Add a `console.log` and inside that `console.log` call the `addTwoAndSeven` function.
```
Comment (Sembauke) [inline curriculum/challenges/english/25-front-end-development/workshop-calculator-app/66cc0a9e06e00b75d6782be9.md:23]:
```suggestion
Add a `console.log` statement, and inside it, call the `addTwoAndSeven` function.
```

File: curriculum/challenges/english/25-front-end-development/workshop-calculator-app/66cc0de2920bec775f610424.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+Even though the functions work as expected, there is a lot of repetition in your code. 
+
+Since you are building a calculator, you don't want to have to create a function for every possible combination of numbers you want to add together.
+
+It would be better to create a single reusable function that can add any two numbers together.
+
+Remove all of the code you have written so far.
+
+In the next step, you will review how to work with function parameters and arguments.
```
Comment (Sembauke) [inline curriculum/challenges/english/25-front-end-development/workshop-calculator-app/66cc0de2920bec775f610424.md:18]:
```suggestion
In the next step, you will review how to work with parameters and arguments in functions.
```

File: curriculum/challenges/english/25-front-end-development/workshop-calculator-app/66cc12fa504b0479dac479a0.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
+id: 66cc12fa504b0479dac479a0
+title: Step 8
+challengeType: 1
+dashedName: step-8
+---
+
+# --description--
+
+The advantage of functions is that they can be called with different arguments. This allows you to reuse the same code with different values.
```
Comment (Sembauke) [inline curriculum/challenges/english/25-front-end-development/workshop-calculator-app/66cc12fa504b0479dac479a0.md:10]:
```suggestion
The advantage of functions is that they can be called with different arguments, allowing you to reuse the same code with various values.
```

File: curriculum/challenges/english/25-front-end-development/workshop-calculator-app/66cc14ca14e65e7a9ebecb80.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+dashedName: step-10
+---
+
+# --description--
+
+Now that your calculator can add two numbers together, it is time to create a function that will subtract two numbers.
+
+Declare a function called `calculateDifference` that takes two parameters, `num1` and `num2`.
+
+Inside the function, return the difference of the two parameters.
```
Comment (Sembauke) [inline curriculum/challenges/english/25-front-end-development/workshop-calculator-app/66cc14ca14e65e7a9ebecb80.md:14]:
```suggestion
Inside the function, return the difference between the two parameters.
```

File: curriculum/challenges/english/25-front-end-development/workshop-calculator-app/66cc14ca14e65e7a9ebecb80.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+assert.isFunction(calculateDifference);
+```
+
+Your `calculateDifference` function should have two parameters called `num1` and `num2`.
+
+```js
+assert.match(code, /calculateDifference\s*(?:=\s*)?\(\s*num1\s*,\s*num2\s*\)\s*/);
+```
+
+Your `calculateDifference` function should return the difference of the two parameters.
```
Comment (Sembauke) [inline curriculum/challenges/english/25-front-end-development/workshop-calculator-app/66cc14ca14e65e7a9ebecb80.md:30]:
```suggestion
Your `calculateDifference` function should return the difference between the two parameters.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

**Review by Sembauke — APPROVED** [review]

**Review by moT01 — COMMENTED** [review]
Nicely done @jdwilkin4 🎉 

Only suggestion I have would be to remove "app" from the dashed names.

**Review by moT01 — APPROVED** [review]

**Pull request merged**